test/blocks/global-footer/global-footer.test.js

```diff
@@ -0,0 +1,266 @@
+        const targetEl = document.querySelector(featuredProductsSelectors[selectorKey]);
+        expect(!!targetEl).to.equal(true);
+      }
+    });
+
+    it('should have social links', async () => {
+      const footer = await createFullGlobalFooter({ waitForDecoration: true });
+
+      expect(footer).to.exist;
+
+      const socialSelectorsKeys = Object.keys(socialLinksSelectors);
+      for (const selectorKey of socialSelectorsKeys) {
+        const targetEl = document.querySelector(socialLinksSelectors[selectorKey]);
+        expect(!!targetEl).to.equal(true);
+      }
+    });
+
+    it('should have region picker', async () => {
+      const footer = await createFullGlobalFooter({ waitForDecoration: true });
+
+      expect(footer).to.exist;
+
+      const regionPickerSelectorsKeys = Object.keys(regionPickerSelectors);
+      for (const selectorKey of regionPickerSelectorsKeys) {
+        const targetEl = document.querySelector(regionPickerSelectors[selectorKey]);
+        expect(!!targetEl).to.equal(true);
+      }
+    });
+
+    it('should have legal section', async () => {
+      const footer = await createFullGlobalFooter({ waitForDecoration: true });
+
+      expect(footer).to.exist;
+
+      const legalSectionSelectorsKeys = Object.keys(legalSelectors);
+      for (const selectorKey of legalSectionSelectorsKeys) {
+        const targetEl = document.querySelector(legalSelectors[selectorKey]);
+        expect(!!targetEl).to.equal(true);
+      }
+    });
+
+    it('should render the whole footer', async () => {
+      const footer = await createFullGlobalFooter({ waitForDecoration: true });
+      expect(footer).to.exist;
+
+      for (const selectorKey of allSelectorsKeys) {
+        const targetEl = document.querySelector(selectors[selectorKey]);
+        expect(!!targetEl).to.equal(true);
+      }
+    });
+  });
+
+  describe('conditional render tests', () => {
+    it('should render the footer when in viewport', async () => {
+      const footer = await createFullGlobalFooter({ waitForDecoration: true });
+
+      expect(footer).to.exist;
+
+      for (const selectorKey of allSelectorsKeys) {
+        const targetEl = document.querySelector(selectors[selectorKey]);
+        expect(!!targetEl).to.equal(true);
+      }
+
+      const elementsAreVisible = await allElementsVisible(
+        selectors,
+        document.querySelector(containerSelector),
+      );
+      expect(elementsAreVisible).to.equal(true);
+    });
+
+    it('should render the footer after 3s when outside of the 300px range of the viewport', async () => {
+      const viewportHeight = window.innerHeight;
+      insertDummyElementOnTop({ height: viewportHeight + 400 });
+
+      const footer = await createFullGlobalFooter({ waitForDecoration: false });
+
+      expect(footer).to.exist;
+
+      for (const selectorKey of allSelectorsKeys) {
+        expect(!!document.querySelector(selectors[selectorKey])).to.equal(
+          false,
```

Comment:
not sure why you split these expect statements on to new lines. same with 170,186.202. Does the `,` necessitate it?

Reply:
I'm using `forEach` now, so we can say this is solved. It was because of my VScode formatter extension